src/mod_activation.f90

```diff
@@ -16,94 +16,86 @@ module mod_activation
   public :: tanhf, tanh_prime
 
   interface
+
     pure function activation_function(x)
       import :: rk
```

Comment:
@milancurcic I just edited my last comment to clarify the first sentence, which was garbled while multitasking.

Reply:
All clear, thanks!

Reply:
@milancurcic Is there an even newer `ifort` that accepts the code?  After years trying to find a Fortran subset that works with multiple versions of each compiler, I concluded it was a time-consuming endeavor that too often ends with cumbersome workarounds or giving up and writing to standards that are decades old.  It's a vicious cycle: the poor support for modern Fortran is partly due to too few programmers submitting bug reports on modern features, which leads to vendors not seeing much demand for modern features.  After several years of submitting on the order of 50 bug reports annually, I found that the compiler support stabilized for the features that I cared most about so I adopted a policy of only supporting the latest version of each compiler unless there was a really strong motivation to do otherwise.  Now that most compilers are free, including the Intel compilers, the argument for supporting older versions is much less compelling than it once was.

Reply:
As far as I can tell, this is the latest release (https://www.intel.com/content/www/us/en/developer/articles/release-notes/oneapi-fortran-compiler-release-notes.html). I tried with ifx 2022.1 just in case, same error.